combat-trainer.lic

```diff
@@ -1910,14 +1919,38 @@ class GameState
     @use_weak_attacks = false
   end
 
+  def use_analyze_flame?
+    @use_analyze_flame
+  end
+
+  def stop_analyze_flame
+    @use_analyze_flame = false
+  end
+
   def melee_attack_verb
     if use_weak_attacks?
       brawling? ? 'punch' : 'jab'
+    elsif use_analyze_flame?
+      update_analyze_flame_array if @analyze_flame_array.empty?
+      @analyze_flame_array.shift
```

Comment:
I was originally worried about balance and position, but now I think it will be ok as-is:
* You'll occasionally see "positive" moves in your combos
* You'll likely end up doing some bob/weave/circle occasionally to regen fatigue
* You can now put up a balance buff with the excess Inner Fire

The log you posted looks like you were fine.